algorand/go-algorand
# Onetechnical/relstable2.1.4

## Summary

## Highlights

- Fix for "ran out of time for round" issue
- Memory usage improvements
- Fix for ARM32 memory panic
- Other tests and fixes

## Changes

1. General
    * New Feature - add simple support for Merkle trees over dense arrays
    * Enhancement - update golang version to latest 1.14.7
    * Bug Fixes
        * Fix 64bit alignment for feePerByte causing memory panic error on ARM32
        * Avoid "ran out of time for round" issue
        * Fix accounting bug in page loading
1. Config
    * Enhancement - update CatchpointInterval config comment
1. KMD
    * Bug Fix - decrease mutex critical section time in kmd CreateWallet to fix rare deadlock on slow systems (#1402)
1. Network
    * New Feature - add an API for using MsgOfInterest messages
1. Tools
    * New Feature - add Rekeying support to pingpong utility
1. Tests
    * Enhancements
        * Add e2e tests for the /v2/assets/{id} endpoint
        * Add a test for goal app --app-account flag
        * Add expect test that combines teal and stateful teal apps in a single group transaction
        * Adding test for parseStringLiteral cases \r and \\\\
1. Other
    * Bug Fixes
        * Allow algorand-indexer to be installed alongside other packages
        * Exclude "beta" dir from sync to s3
        * Dynamically install deb package in legacy pipeline
        * check return code of dpkg-query show-format query in apt preinstall script

## Test Plan

Standard release testing.
